project/src/main/java/com/google/sps/data/User.java

```diff
@@ -28,9 +30,15 @@
   /** The idap of the user */
   private String userIdap;
 
+  private List<UserCompetitionInfo> allCompetitions = new ArrayList<>();
+
   public User(long id, String userName, String userIdap) {
     this.id = id;
     this.userName = userName;
     this.userIdap = userIdap;
   }
+
+  public void addCompetition(UserCompetitionInfo competition) {
```

Comment:
Maybe error check here to make sure this competition isn't already in the List, or use a set instead